pkg/catalogv2/content/content.go

```diff
@@ -179,6 +179,21 @@ func (c *Manager) Icon(namespace, name, chartName, version string) (io.ReadClose
 		return nil, "", err
 	}
 
+	// Check if the system catalog is bundled which means we are at an airgapped environment.
+	// If so, check if the icons are available locally.
+	// If the icons are not available locally, return a nil io.ReadCloser
+	if settings.SystemCatalog.Get() == "bundled" && isRancherCatalog(name) {
+		// If the chart icon is not an HTTP URL and the repository has a commit status,
+		// attempt to get the icon from the git repository.
+		if !isHTTP(chart.Icon) && repo.status.Commit != "" {
```

Comment:
This if condition is being duplicated at line no 197. 

So, maybe we can change the logic like this 

```
if !isHTTP(chart.Icon) && repo.status.Commit != "" {
		return git.Icon(namespace, name, repo.status.URL, chart)
	}

if settings.SystemCatalog.Get and ranchercatalog {

return  nil ... 

}
```

Reply:
Resolved